quay_search.py

```diff
@@ -0,0 +1,135 @@
+        parameters = {'public':'true','namespace': self._organization}
+        r = requests.get(url, headers={'Accept-encoding': 'gzip'}, params=parameters,
+                            timeout=12)
+
+        # delete old index data and create the dir "indexdir" new
+        if os.path.exists("indexdir"):
+            shutil.rmtree("indexdir")
+        os.mkdir("indexdir")
+        schema = Schema(title=TEXT(stored=True), content=STORED)
+        self._index = create_in("indexdir", schema)
+
+        json_decoder = json.JSONDecoder()
+        decoded_request = json_decoder.decode(r.text)
+        writer = self._index.writer()
+        for i in xrange(len(decoded_request['repositories'])):
+            writer.add_document(title=decoded_request['repositories'][i]['name'], content=decoded_request['repositories'][i]['description'])
+        writer.commit()
+
+    def search_repository(self, p_search_string, p_non_strict):
+        """This function searches matching repositories of an organization.
+         Results are displayed with all available versions. Docker download command is given too."""
+        # with statement closes searcher after usage.
+        with self._index.searcher() as searcher:
+            search_string = u"*"
+            search_string += p_search_string
+            search_string += "*"
+            query = QueryParser("title", self._index.schema).parse(search_string)
+
+            results = searcher.search(query)
+            
+            if p_non_strict:
+                # look for spelling errors and use suggestions as a search term too
+                corrector = searcher.corrector("title")
+                alternative_strings =  corrector.suggest(p_search_string, limit=2)
+
+                # get all repositories with suggested keywords
+                for i in alternative_strings:
+                    search_string = u"*"
+                    search_string += i
+                    search_string += "*"
+                    query = QueryParser("title", self._index.schema).parse(search_string)
+                    results_tmp = searcher.search(query)
+                    results.extend(results_tmp)
+           
+            # get all versions for the found tools
+            dict_results = {}
+            for i in results:
+                additional_info = self.get_additional_repository_information(i['title'])
+                versions = []
+                for j in additional_info:
+                    versions.append(j)
+                dict_results[i['title']] = versions
+            
+            print "The query ", '\033[1m' + p_search_string + '\033[0m', " resulted in", len(results) ,"result(s).",
+            
+            if p_non_strict:
+                print "The search lists the results for ",
+                for i in xrange(len(alternative_strings)):
+                    print '\033[1m' + alternative_strings[i] + '\033[0m', ", ",
+                print "too."
+            for i in dict_results:
```

Comment:
It's not C and you should use the features a language provides to you to make code faster, more readable and more robust, isn't it?